frontend/components/common/InstantSearchProvider.tsx

```diff
@@ -169,7 +169,16 @@ export function InstantSearchProvider({
                }
             );
 
-            return `${baseUrl}${path}${queryString}`;
+            let customQueries = '';
+            const customQueryParams = new URLSearchParams(location.search);
+            if (customQueryParams.has('_vercel_no_cache')) {
+               customQueries =
+                  (queryString ? '&' : '?') +
+                  '_vercel_no_cache=' +
+                  customQueryParams.get('_vercel_no_cache');
+            }
+
+            return `${baseUrl}${path}${queryString}${customQueries}`;
```

Comment:
Is there a reason that this isn't integrated into the `qsModule.stringify` call above? Something like this might work:
```js
            const queryParams = new URLSearchParams(location.search);
            const cacheParamName = '_vercel_no_cache'
            const cacheParam = queryParams.get(cacheParamName) || undefined;

            const queryString = qsModule.stringify(
               { ...routeState, filter: filterCopy, [cacheParamName]: cacheParam },
               {
                  addQueryPrefix: true,
                  arrayFormat: 'indices',
               }
            );
```

[These seem to be the `stringify` docs](https://github.com/ljharb/qs#stringifying); found via [Algolia's docs](https://www.algolia.com/doc/guides/building-search-ui/going-further/routing-urls/react/#createurlsearchstate).

Reply:
> Properties that are set to undefined will be omitted entirely

I like this better than my suggestion!